engine/map_test.go

```diff
@@ -20,15 +26,69 @@ print(m)
 
 	// test map<string,int>
 	stdout, err = runScript(nil, "", `
-m = datalark.Map({"a": 123})
-print(m)
-`)
+		m = datalark.Map({"a": 123})
+		print(m)
+	`)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertEqual(t, stdout, `map{
 	string{"a"}: int{123}
 }
 `)
+}
 
+// Test that a map in a struct gives the correct number of fields
+func TestMapInStructFields(t *testing.T) {
+	// Parse schema
+	typesystem, err := ipld.LoadSchema("<noname>", strings.NewReader(`
+		type LookupTable struct {
+			name String
+			data Map[String]String
```

Comment:
You're trying to use golang syntax here :P

You want `data {String:String}`.

(It's a little wild that this ended up getting parsed the way it did!  I guess, uh... the DSL parser doesn't require whitespace between the end of the last field declaration, and the start of the next one... and also nothing's enforcing field name character range restrictions yet.  Whoops.)

Reply:
Oh wow, so it sees `data Map` `[ String` `] String`, since the parser is being too permissive?

Reply:
It appears I also need to change

```
	type LookupTable struct {
		Name string
		Data map[string]string
	}```

to this:

```
	type LookupTable struct {
		Name string
		Data struct{
			Keys []string
			Values map[string]string
		}
	}
```

So a native golang map doesn't work, it needs to be this Keys/Values struct. Is that intentional or a missing feature?

Reply:
A little column A, a little column B :cry: 

This Keys+Values struct is so it can preserve order.

I think ideally we'd like to use Golang's new generics features to be able to support more elegant ways of solving this problem.  (I don't know exactly how that would work, though.  It's just a directional wish.)